Makefile

```diff
@@ -325,23 +334,63 @@ release-qa: require-tag require-gh-token require-s3cmd require-ruby
-	echo "Uploading OSX binary for auto-updates" && \
-	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_darwin_amd64.bz2 && \
-	echo "Uploading Linux binaries for auto-updates" && \
-	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_linux_amd64.bz2 && \
-	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_linux_386.bz2
+	for NAME in $$(ls -1 update_darwin_amd64.bz2 update_linux_386.bz2 update_linux_amd64.bz2 update_windows_386.bz2); do \
+		echo "Copying update binary $$NAME..." && \
+		$(S3CMD) put -P $$NAME s3://$(S3_BUCKET); \
+	done && \
+	git tag -a "$$TAG" -f --annotate -m"Tagged $$VERSION" && \
+	git push --tags -f
 
-release-beta:
+release-beta: require-s3cmd
 	@BASE_NAME="lantern-installer-qa" && \
 	BETA_BASE_NAME="lantern-installer-beta" && \
-	for NAME in $$(ls -1 $$BASE_NAME.exe $$BASE_NAME.dmg $$BASE_NAME-32-bit.deb $$BASE_NAME-64-bit.deb); do \
+	for URL in $$($(S3CMD) ls s3://$(S3_BUCKET)/ | grep $$BASE_NAME | awk '{print $$4}'); do \
+		NAME=$$(basename $$URL) && \
 		BETA=$$(echo $$NAME | sed s/"$$BASE_NAME"/$$BETA_BASE_NAME/) && \
-		echo "Copying alpha $$NAME to beta $$BETA..." && \
 		$(S3CMD) cp s3://$(S3_BUCKET)/$$NAME s3://$(S3_BUCKET)/$$BETA; \
 	done
 
+release: require-tag require-s3cmd require-gh-token require-wget require-ruby require-lantern-binaries
+	@TAG_COMMIT=$$(git rev-list --abbrev-commit -1 $$TAG) && \
+	if [[ -z "$$TAG_COMMIT" ]]; then \
+		echo "Could not find given tag $$TAG."; \
+	fi && \
+	BASE_NAME="lantern-installer-beta" && \
+	PROD_BASE_NAME="lantern-installer" && \
+	for URL in $$($(S3CMD) ls s3://$(S3_BUCKET)/ | grep $$BASE_NAME | awk '{print $$4}'); do \
+		NAME=$$(basename $$URL) && \
+		PROD=$$(echo $$NAME | sed s/"$$BASE_NAME"/$$PROD_BASE_NAME/) && \
+		$(S3CMD) cp s3://$(S3_BUCKET)/$$NAME s3://$(S3_BUCKET)/$$PROD; \
+	done && \
+	for URL in $$($(S3CMD) ls s3://$(S3_BUCKET)/ | grep update_ | awk '{print $$4}'); do \
+		NAME=$$(basename $$URL) && \
+		$(S3CMD) get --force s3://$(S3_BUCKET)/$$NAME $$NAME; \
+	done && \
+	$(RUBY) ./installer-resources/tools/createrelease.rb "$(GH_USER)" "$(GH_RELEASE_REPOSITORY)" $$TAG && \
+	echo "Uploading Windows binary for auto-updates" && \
+	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_windows_386.bz2 && \
+	echo "Uploading OSX binary for auto-updates" && \
+	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_darwin_amd64.bz2 && \
+	echo "Uploading Linux binaries for auto-updates" && \
+	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_linux_amd64.bz2 && \
+	$(RUBY) ./installer-resources/tools/uploadghasset.rb $(GH_USER) $(GH_RELEASE_REPOSITORY) $$TAG update_linux_386.bz2 && \
+	echo "Copying binaries to $(LANTERN_BINARIES_PATH)..." && \
```

Comment:
Pulling binaries from S3 and uploading them to https://github.com/getlantern/lantern-binaries.